scalanet/src/io/iohk/scalanet/peergroup/dynamictls/CustomTlsValidator.scala

```diff
@@ -85,22 +127,24 @@ object CustomTlsValidator {
 
   /**
     *
-    * Performs all validations required by libp2p spec
+    * Certificates MUST use the NamedCurve encoding for elliptic curve parameters.
+    * Endpoints MUST abort the connection attempt if is not used.
     *
     */
-  def validateCertificates(
-      certificates: Array[X509Certificate],
-      connectingTo: Option[BitVector]
-  ): Either[CertificateError, SignedKey] = {
-    for {
-      cert <- validateCertificatesQuantity(certificates)
-      _ <- validateCertificateDate(cert)
-      extension <- validateCertificateExtension(cert, SignedKey.extensionIdentifier)
-      signedKeyExtension <- validateExtension(extension)
-      _ <- validateCertificateSignature(signedKeyExtension, cert)
-      _ <- if (connectingTo.isDefined) validateServerId(signedKeyExtension, connectingTo.get) else Right(())
-    } yield {
-      signedKeyExtension
+  def validateCertificatePublicKey(cert: X509Certificate): Either[CertificateError, ECPublicKey] = {
+    val certificatePublicKey = cert.getPublicKey
+    Try {
+      val ecPublicKey = certificatePublicKey.asInstanceOf[ECPublicKey]
+      val subjectKeyInfo = SubjectPublicKeyInfo.getInstance(ASN1Primitive.fromByteArray(ecPublicKey.getEncoded))
+      val x962Params = X962Parameters.getInstance(subjectKeyInfo.getAlgorithm.getParameters)
+      (ecPublicKey, x962Params)
+    }.toEither.left.map(_ => WrongCertificateKeyFormat).flatMap {
+      case (key, parameters) =>
+        if (parameters.isNamedCurve) {
+          Right(key)
+        } else {
+          Left(WrongCertificateKeyFormat)
+        }
     }
```

Comment:
```suggestion
    } match {
      case Success((key, parameters)) if parameters.isNamedCurve =>
        Right(key)
      case _ =>
          Left(WrongCertificateKeyFormat)
    }
```
If you wanted to match the style of the other `Try` constructs.